### ISSUE TITLE
Prune unnecessary etcd goroutines

### ISSUE THREAD

**Issue by joe-elliott** [grafana/tempo#28]
tempo starts up etcd goroutines even when not using the etcd client.  Prevent these from being started.  Possibly requires a PR to Cortex.

```
goroutine 57 [chan receive]:
go.etcd.io/etcd/pkg/logutil.(*MergeLogger).outputLoop(0xc0001c9b40)
	/home/joe/dev/grafana/frigg/vendor/go.etcd.io/etcd/pkg/logutil/merge_logger.go:173 +0x3a3
created by go.etcd.io/etcd/pkg/logutil.NewMergeLogger
	/home/joe/dev/grafana/frigg/vendor/go.etcd.io/etcd/pkg/logutil/merge_logger.go:91 +0x80

goroutine 106 [chan receive]:
go.etcd.io/etcd/pkg/logutil.(*MergeLogger).outputLoop(0xc0001f0a60)
	/home/joe/dev/grafana/frigg/vendor/go.etcd.io/etcd/pkg/logutil/merge_logger.go:173 +0x3a3
created by go.etcd.io/etcd/pkg/logutil.NewMergeLogger
	/home/joe/dev/grafana/frigg/vendor/go.etcd.io/etcd/pkg/logutil/merge_logger.go:91 +0x80

goroutine 109 [chan receive]:
go.etcd.io/etcd/pkg/logutil.(*MergeLogger).outputLoop(0xc0001f0c00)
	/home/joe/dev/grafana/frigg/vendor/go.etcd.io/etcd/pkg/logutil/merge_logger.go:173 +0x3a3
created by go.etcd.io/etcd/pkg/logutil.NewMergeLogger
	/home/joe/dev/grafana/frigg/vendor/go.etcd.io/etcd/pkg/logutil/merge_logger.go:91 +0x80
```

**Comment by mdisibio** [comment]
This is worth doing because it makes many goroutines (dozens-100+), may need more research.

**Comment by 56quarters** [comment]
This can likely be closed due to #969
